doc/howto/dqlite_troubleshoot.md

```diff
@@ -0,0 +1,85 @@
+# How to troubleshoot (some) dqlite errors
+
+dqlite is the distributed database that LXD uses to store information
+that must be synchronized across a cluster. This page describes how
+to respond if LXD fails to start or crashes because of a dqlite-related error.
+
+## Recognizing dqlite-related errors
+
+Look for error messages that mention keywords like `dqlite`, `raft`, or
+`segment`, especially if LXD is failing to start up.
+
+A known risk factor for some of the errors covered below is a previous LXD
+crash caused by running out of disk space.
+
+## The dqlite data directory
+
+When investigating dqlite-related errors, it's essential to look at the
+contents of the dqlite data directory for the affected node. This is the
+directory where the local instance of dqlite stores all its data. If you
+installed the LXD snap, you can find this directory at
+`/var/snap/lxd/common/lxd/database/global`.
+
+The data directory contains several types of file. The most important types are:
+
+- Closed segments: these have filenames like
+  `0000000000056436-0000000000056501`. The two numbers are the *start index*
+   and *end index*. Both indices are inclusive.
+- Open segments: these have filenames like `open-1`
+- Snapshot files: these have names like `snapshot-1-59392-27900`. The first
+  number is the *snapshot index*.
+- Snapshot metadata files: these have names like `snapshot-1-59392-27900.meta`
+  and are paired with snapshot files.
+
+## Spotting weirdness
+
+When looking at the contents of the data directory, watch for the following
+symptoms:
+
+1. Closed segments whose index ranges overlap (remember that these ranges are inclusive)
+2. A closed segment with end index X where the next closed segment has start index greater than X + 1
+3. A snapshot file with snapshot index X where the next closed segment has start index greater than X + 1
+4. A snapshot file whose size is less than the size of a previous (lower-numbered) snapshot
```

Comment:
```suggestion
- Closed segments whose index ranges overlap (remember that these ranges are inclusive).
- A closed segment with end index X where the next closed segment has start index greater than X + 1.
- A snapshot file with snapshot index X where the next closed segment has start index greater than X + 1.
- A snapshot file whose size is less than the size of a previous (lower-numbered) snapshot.
```

I'm actually not sure why the linter complains about the ordered list prefixes - but is there a reason to use an ordered list here? This seems like a list of things to look out for, no matter in which order.

Reply:
The reason is just to be able to refer unambigously (by number) to the symptoms in the next section. I could use an unordered list and just say put "(symptom A)" (etc.) after each item?

Reply:
It's a weird markdownlint quirk, see https://github.com/canonical/lxd/pull/13346/commits/df765b97e69d6dc7a4cbb4c43f64c03a6843f283